Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -356,6 +356,13 @@ void RenderBox::styleDidChange(StyleDifference diff, const RenderStyle* oldStyle
         }
     }
 
+#if ENABLE(DARK_MODE_CSS)
+    if (layer() && oldStyle && oldStyle->colorScheme() != newStyle.colorScheme()) {
```

Comment:
I wonder if there are other things we need to update on a `color-scheme` change? Are system colors (used as a background / text color) updating correctly here?

This solution is fine, but I wonder if there's something existing we can hook into – I need to think a bit.

Reply:
`data:text/html,<p>Hello</p><style>:root%20{color-scheme:light;}</style><script>setTimeout(()=>document.documentElement.style.colorScheme="dark",500);</script>`

From a quick test using the above URL the system colours do seem to be updating.

Interestingly the viewport level scrollbars do update (excepting the scroll corner), so not sure why non viewport scrollers don't get invalidated correctly. Perhaps the viewport ones get invalidated as a side effect

Reply:
Yeah, that test case will work because of the `recalculateBaseBackgroundColor` call for the root element.

Would be more interesting to change the `color-scheme` of a specific element.